### PR TITLE
[RFC] Enable automatic closing of stale issues and pull requests by github actions.

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

I thought that issues / PRs (especially issues) that were left without updates for a certain amount of time could be closed automatically, because

1. If you don't want them to be stale, the `no-stale` label is attached.
2. I feel that the number of issues with stale is increasing.

## Description of the changes

Change the configuration of the stale action.

### CODE REVIEW

**Comment by crcrpar** [comment]
Could you tell me why 14days?

**Comment by g-votte** [comment]
I think the period of `14` days is a bit short, since many issues/PRs take more than that. For example, `100` or `200` would be much safer.

**Comment by HideakiImamura** [comment]
@hvy @toshihikoyanase @himkt @c-bata Please feel free to leave comments if you have.

**Comment by HideakiImamura** [comment]
I agree with @g-votte to be `100` of the period. 2 weeks is too short.

**Comment by keisuke-umezawa** [comment]
Thank you for your comments. I modified the number from 14 to 100.

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -16,8 +16,10 @@ jobs:
...
         stale-issue-message: 'This issue has not seen any recent activity.'
         stale-pr-message: 'This pull request has not seen any recent activity.'
+        close-issue-message: 'This issue was closed automatically because it had not seen any recent activity. If you want to discuss it, you can reopen it freely.'
+        close-pr-message: 'This pull request was closed automatically because it had not seen any recent activity. If you want to discuss it, you can reopen it freely.'
         days-before-stale: 14
-        days-before-close: -1
+        days-before-close: 100
         stale-issue-label: 'stale'
         stale-pr-label: 'stale'
         exempt-issue-labels: 'no-stale'
```
Comment (g-votte) [inline .github/workflows/stale.yml:25]: How about adding 'contribution-welcome' here?

**Comment by HideakiImamura** [comment]
According to the internal discussion, I think we should close both issues and PRs in 100 days as a deadline. And, I think https://github.com/optuna/optuna/pull/2287#discussion_r571757356 is reasonable. Could you apply it?

**Comment by keisuke-umezawa** [comment]
@HideakiImamura 
> According to the internal discussion, I think we should close both issues and PRs in 100 days as a deadline. And, I think #2287 (comment) is reasonable. Could you apply it?

Sorry, but I already applied it by this commit https://github.com/optuna/optuna/pull/2287/commits/6b36d7ac52e5be86b2f3adc2a3158009cd502803 .

**Review by crcrpar — APPROVED** [review]

**Review by g-votte — APPROVED** [review]
LGTM

**Pull request merged**